data-processing/entities/sentences_pdf/commands/fetch_spp_jsons.py

```diff
@@ -0,0 +1,76 @@
+import os
+
+import json
+import time
+from argparse import ArgumentParser
+import logging
+from typing import Iterator, Optional
+from dataclasses import dataclass
+
+from common import file_utils
+from common import directories
+from common.commands.base import ArxivBatchCommand
+from common.fetch_arxiv import FetchFromArxivException, fetch_from_arxiv, fetch_from_s3
+from common.types import ArxivId, RelativePath
+
+
+import requests
+
+#  TODO - dont need
+# DEFAULT_S3_ARXIV_SOURCES_BUCKET = "s2-arxiv-sources"
+
+"""Constants for dealing with retries and delays between call attempts"""
+DEFAULT_FETCH_DELAY = 10  # seconds
+BACKOFF_FETCH_DELAY = 60  # seconds
+MAX_FETCH_ATTEMPTS = 3
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class SppJsonTask:
+    arxiv_id: ArxivId
+    pdf_path: RelativePath
+    spp_host: str
+
+
+class FetchSppJsons(ArxivBatchCommand[ArxivId, None]):
+    @staticmethod
+    def get_name() -> str:
+        return "fetch-spp-jsons"
+
+    @staticmethod
+    def get_description() -> str:
+        return "Send a PDF to SPP to get back a JSON"
+
+    def get_arxiv_ids_dirkey(self) -> Optional[str]:
+        return 'arxiv-pdfs'
+
+    def load(self) -> Iterator[SppJsonTask]:
+        for arxiv_id in self.arxiv_ids:
+            yield SppJsonTask(
+                arxiv_id=arxiv_id,
+                pdf_path=os.path.join(directories.arxiv_subdir('arxiv-pdfs', arxiv_id=arxiv_id), f'{arxiv_id}.pdf'),
```

Comment:
So for this, is this what we had chatted about one day, where we let scholarphi compile the Latex->PDF, and we pick up the local PDF here?